### DISCUSSION TITLE
Is there an existing set of scripts and ref scripts live on preprod?

### DISCUSSION THREAD

**Discussion by HinsonSIDAN** [Anastasia-Labs/single-asset-staking-offchain#13] [Q&A]
Integrating with maestro endpoint and want to conduct some testing, wonder if there is existing set of scripts live in preprod testnet so to skip the heavy initialization part. Integration on endpoints:
- `stakeToken`
- `claimReward`
- `updateStake`

If so, also is there an exact mapping of which compiled script on the on-chain repo corresponding to the input expected by the offchain endpoints. Just to make sure there is no unintended error as the names on config param and the names of compiled scripts are not of exact match.

PR of the implementation for reference:
https://github.com/maestro-org/midi/pull/10

**Comment by nikhils9** [comment]
Currently there aren't live ref scripts on preprod.

We are currently working on completely eliminating the elaborate initialization setup. So that the deployed ref scripts are reusable by all the different staking campaigns. And we take care of deploying the ref scripts just once beforehand. Making it convenient and cheaper for the end users of the APIs. We anticipate to complete these changes next week, then proceed with deploying the ref scripts on preprod first, performing testing on our side and then subsequently on mainnet.

>If so, also is there an exact mapping of which compiled script on the on-chain repo corresponding to the input expected by the offchain endpoints. Just to make sure there is no unintended error as the names on config param and the names of compiled scripts are not of exact match.

That's a valid concern. The current naming and usage of compiled scripts can get ambiguous. Since we are anyways making updates, I'll get the compiled scripts name to be uniform across the on-chain and off-chain repos. I hope that'll alleviate current concerns of missing/incorrect scripts.

**Reply by HinsonSIDAN** [reply]
Thanks! Please let me know if there is update on this, so the public api could be progressed.

**Reply by nikhils9** [reply]
Sure, I'll update here.
